pkgs/development/node-packages/generate.sh

```diff
@@ -1,10 +1,15 @@
 #!/usr/bin/env nix-shell
-#! nix-shell -I nixpkgs=../../.. -i bash -p nodePackages.node2nix
-# NOTE: Script must be run from the node-packages directory
-
+#! nix-shell shell-generate.nix -i bash
 set -eu -o pipefail
 
-rm -f node-env.nix
-node2nix --nodejs-10 -i node-packages-v10.json -o node-packages-v10.nix -c composition-v10.nix
-node2nix --nodejs-12 -i node-packages-v12.json -o node-packages-v12.nix -c composition-v12.nix
-node2nix --nodejs-13 -i node-packages-v13.json -o node-packages-v13.nix -c composition-v13.nix
+cd "$NODE_NIXPKGS_PATH/pkgs/development/node-packages"
```

Comment:
Did you know you can have the current path in nix?

`let here = toString ./.;`

You don’t have to do any tmpdir/cd magic, just use `here` as an absolute path.

Reply:
try it out with `nix-instantiate --eval -E 'toString ./.'`.

Reply:
the tmpdir magic is in order to not break nixpkgs if something goes wrong  during evaluation (which happens if your network is unstable for instance)

Reply:
I’d argue that’s what git is for, but I’m fine with the code as it stands.